src/api/__tests__/index.spec.js

```diff
@@ -0,0 +1,272 @@
+import FreshDataApi from '../index';
+
+describe( 'api', () => {
+	describe( '#constructor', () => {
+		it( 'should initialize clients map', () => {
+			const api = new FreshDataApi();
+			expect( api.clients ).toEqual( new Map() );
+		} );
+
+		it( 'should initialize state', () => {
+			const api = new FreshDataApi();
+			expect( api.state ).toEqual( {} );
+		} );
+
+		it( 'should initialize dataHandlers', () => {
+			const api = new FreshDataApi();
+			expect( api.dataHandlers ).toEqual( {} );
+		} );
+
+		it( 'should use api methods defined in subclass', () => {
+			const methods = { get: () => {} };
+			class MyApi extends FreshDataApi {
+				static methods = methods;
+			}
+			const api = new MyApi();
+			expect( api.methods ).toBe( methods );
+		} );
+
+		it( 'should use endpoint methods defined in subclass', () => {
+			const endpoints = { things: { read: () => {} } };
+			class MyApi extends FreshDataApi {
+				static endpoints = endpoints;
+			}
+			const api = new MyApi();
+			expect( api.endpoints ).toBe( endpoints );
+		} );
+
+		it( 'should use selector methods defined in subclass', () => {
+			const selectors = { getThings: () => () => {} };
+			class MyApi extends FreshDataApi {
+				static selectors = selectors;
+			}
+			const api = new MyApi();
+			expect( api.selectors ).toBe( selectors );
+		} );
+	} );
+
+	describe( '#setDataHandlers', () => {
+		it( 'should set dataHandlers', () => {
+			const dataRequested = jest.fn();
+			const dataReceived = jest.fn();
+			const errorReceived = jest.fn();
+
+			class MyApi extends FreshDataApi {
+			}
+			const api = new MyApi();
+			api.setDataHandlers( dataRequested, dataReceived, errorReceived );
+
+			expect( api.dataHandlers.dataRequested ).toBe( dataRequested );
+			expect( api.dataHandlers.dataReceived ).toBe( dataReceived );
+			expect( api.dataHandlers.errorReceived ).toBe( errorReceived );
+		} );
+	} );
+
+	describe( '#createClient', () => {
+		it( 'should create a client', () => {
+			class MyApi extends FreshDataApi {
+			}
+			const api = new MyApi();
+			const client = api.createClient( 'myKey' );
+
+			expect( client ).toBeInstanceOf( ApiClient );
+			expect( client.key ).toBe( 'myKey' );
+			expect( api.clients.size ).toBe( 1 );
+			expect( api.clients.get( 'myKey' ) ).toEqual( client );
+		} );
+	} );
+
+	describe( '#findClient', () => {
+		it( 'should find an existing client', () => {
+			class MyApi extends FreshDataApi {
+			}
+			const api = new MyApi();
+			const client = api.createClient( 'myKey' );
+			const foundClient = api.findClient( 'myKey' );
+
+			expect( foundClient ).toBe( client );
+		} );
+
+		it( 'should not find an existing client', () => {
+			class MyApi extends FreshDataApi {
+			}
+			const api = new MyApi();
+			const client = api.createClient( 'myKey' );
+			const foundClient = api.findClient( 'myKeey' );
+
+			expect( foundClient ).not.toBe( client );
+			expect( foundClient ).toBeNull();
+		} );
+	} );
+
+	describe( '#getClient', () => {
```

Comment:
In most cases, the code will use `getClient` because it lazy-creates the client.
I've mostly split them out for testability and legibility of code.

There may be a case someday where `findClient` is desired, but that's not really why I added it.